### PR TITLE
 Warning Message for Moving Attached Open Files #10121 

### LINKED ISSUE #10121: Warning Message for Moving Attached Open Files
Suppose that you attach a file to an entry. Then, using a cleanup you want to move that file to the General File Directory of the library. If the attached file is open, the file is not moved but no warning message is shown. It would be nice to have such a message to let the user know what just happened.

### PR DESCRIPTION
**Pull request by HrithikPadavala**
Closes #10121 

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
Untitled

### CODE REVIEW

**Comment by koppor** [comment]
@HrithikPadavala Maybe a picture helps to understand the output of CHANGELOG.md check:

image

File: src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java [hunk 1]
```diff
@@ -24,9 +27,12 @@ public class MoveFilesCleanup implements CleanupJob {
     private final BibDatabaseContext databaseContext;
     private final FilePreferences filePreferences;
 
-    public MoveFilesCleanup(BibDatabaseContext databaseContext, FilePreferences filePreferences) {
+    private final DialogService dialogService;
```
Comment (koppor) [inline src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java:30]: Oh, no `gui` in logic.

Please add a List<String> of errors. Add a getter for it. after the call, from the gui, call the getter in the gui and display (sing the dialogservice)

File: src/main/resources/l10n/JabRef_en.properties [hunk 2]
```diff
@@ -1,4 +1,8 @@
 Unable\ to\ monitor\ file\ changes.\ Please\ close\ files\ and\ processes\ and\ restart.\ You\ may\ encounter\ errors\ if\ you\ continue\ with\ this\ session.=Unable to monitor file changes. Please close files and processes and restart. You may encounter errors if you continue with this session.
+
+Could\ not\ move\ file.\ Please\ close\ all\ the\ attached\ files\ and\ retry.=Could not move file. Please close all the attached files and retry.
```
Comment (koppor) [inline src/main/resources/l10n/JabRef_en.properties:3]: Please include the filename here. Example: If have 100 files attached, 2 are opened, two cause issues, 98 not. Does the user know which file to close?!

**Review by koppor — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java (hunk 1 above)
Comment (HrithikPadavala) [reply src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java:30]: @koppor, so should I create an instance of MoveFilesCleanup in gui now?

File: src/main/java/org/jabref/gui/cleanup/CleanupAction.java [hunk 3]
```diff
@@ -105,9 +105,10 @@ private void doCleanup(BibDatabaseContext databaseContext, CleanupPreferences pr
         CleanupWorker cleaner = new CleanupWorker(
                 databaseContext,
                 preferences.getFilePreferences(),
-                preferences.getTimestampPreferences());
+                preferences.getTimestampPreferences(),
+                dialogService);
 
-        List<FieldChange> changes = cleaner.cleanup(preset, entry);
+            List<FieldChange> changes = cleaner.cleanup(preset, entry);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/cleanup/CleanupAction.java:111]: Please no additgional indent.

File: src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java:30]: You already have. See src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java.

You need to work on passing back the results.

The cleanup should not worry of how to format the results; that should be done by the caller.

**Review by koppor — CHANGES_REQUESTED** [review]

**Comment by koppor** [comment]
Closing this issue due to inactivity :zzz: 

Please ping us if you intend to resume work on this one.

**Comment by HrithikPadavala** [comment]
Hi @koppor , Let me grab another issue which I could tackle this time.

**Comment by ThiloteE** [comment]
@HrithikPadavala Is there a reason you gave up on this PR? Oliver spent some of his precious time to review this PR, so it is sad to see it go into the rubbish bin.

**Pull request closed without merging**